---
layout: post
title:  "DON’T PANIC! A Hitchhiker's Guide to Genomics in Data Portals"
breadcrumb: true
author: rosi_bajari
date: 2018-07-31
categories: rosi_bajari
tags:
    - Genomics
    - Data Portal
    - NGS
    - Next Generation Sequencing
    - Business Analyst
    - Bioinformaticsg
teaser:
    info: Curious about what genomics are, but aren't a bio buff?  This quick guide will get you up to speed.
    image: rosi_bajari/hitchhikers/dont_panic.jpg
header:
    version: small
    title: Software Engineering Blog
    image: header-logo-crop.png
    icon: icon-blog
---

Usually when I say I am a Business Analyst, people think that I work at a bank or with business software. Little do they know, my world is way more exciting than that! As a Business Analyst with OICR, I get to work on some of the most uh-mazing stuff like data portals, genomic data visualization, and software packages that support genomic research. This usually results in questions like _"What's genomics?"_, _"What’s a data portal?"_, or even _"Can you tell me what this medical report means?"_. So, I can’t answer that last one, but I can give a a brief overview of genomics!


## A brief history of genomics 

Way back in the 50s, Oxford University scientists Watson and Crick discovered the 3-D structure of DNA (controversially helped by questionally obtained data from Rosalind Franklin and Maurice Wilkins), and Frederick Sanger sequenced the first protein (insulin) kicking off a scientific frenzy to figure out all the biochemical processes. After these discoveries, DNA research floundered for awhile (they even thought it was useless for awhile), but eventually by the 60s some scientists successfully sequenced RNA, which was easier to do as it is naturally more stable. Then Sanger (yup, that same guy again) developed a method to sequence DNA which was automated in the 1980s. Sanger sequencing was revolutionary but by today's standards pretty slow. By the 90s, the Human Genome Project had started the ambitious goal to sequence the whole human genome, which they successfully did and we celebrated for awhile. But, as scientists do, they got impatient and decided to come up with a solution that was harder, better, faster stronger: massively parallel, high-throughput Next Generation Sequencing (NGS). 

For a more complete brief history read [this](https://www.nature.com/articles/nature24286), or for a more in depth and entertaining history I suggest this great [read](https://www.goodreads.com/book/show/27276428-the-gene).


## So, uh, what did that last thing actually mean!?

  
NGS is probably what people are referring to when they talk about “genomics”. Massively parallel, high-throughput Next Generation Sequencing is sequencing on steroids - It's faster than anything else, cooler than anything else, and produces a *tonne* of data. The basic idea of NGS this: Get someone's DNA, shred it into a lot of pieces, make a lot of copies of it, and then sequence it with a machine. Sounds easy right?

You can think of DNA sequencing like this:

>You need to find a single paragraph in a book. To find the paragraph you are interested in, you first have to shred all its pages. Now you have to glue it back together, but a lot of the paragraphs look really similar since they have words like “the”, “when”, and "too" everywhere. Also, maybe you accidently got some of another books pages mixed in, so you need to clean those out while you sort and glue things together.

  
Hmm, not so easy. How exactly does one accomplish such a monumental task then? With this simple recipe:

1.  Use super cool science to harvest some DNA.
    
2.  Since DNA is so small, we like to increase our chances of success of sequencing it by amplifying it into many copies.
    
3.  Then, it goes into a super fancy sequencing machine that probably operates on a sequencing-by-synthesis methods. Individual base pairs (A-T-C-G) are attached to a special fluorescent dye that a camera reads as each base pair gets individually added to a DNA strand. Remember that amplification in step 2? This is why - the camera might not be able to read the fluorescent signal from one base, but it can with many signal powers combined. At the end, the sequencing machine spits out FASTQ data files that contains “unaligned reads” - small text strings of base pairs.
    
4.  So now you have a (really) giant pile of small strings of DNA that you need to “align” or glue back together into the original DNA strand. People have developed... [a few algoritms](https://en.wikipedia.org/wiki/List_of_sequence_alignment_software) for this. Across the board, there are basic inputs: the FASTA files you just made, and a reference genome. The reference genome is a widely accepted standard template that says _exactly_ where each gene is located. Reads are compared to the reference genome as a template to reconstruct the original genome and output as a new file - a BAM file of “aligned reads”.

Congratulations - Your genome is now sequenced! So what now?

<center>
  <figure >
      <img src="{{site.urlimg}}rosi_bajari/hitchhikers/basic-image.png"/>
  </figure>
</center>

## Spot the Difference!

There’s all kinds of fun stuff you can do! Usually, the first thing that researchers do is try and look for any differences in the genome, called mutations, through a methods like “variant calling”.


There’s a few different aberrations that can appear in DNA that we regularly search for. These are some of the most popular types of DNA differences:

-   SNPs (Single Nucleotide Polymorphisms), pronounced “snips”. This is where a *single* tiny base is changed in the DNA, which [sometimes has no effect and sometimes has a huge effect ](https://www.thisamericanlife.org/577/something-only-i-can-see/act-one).
    
-   Indels: sequences of 1-1000 bases that have been inserted or deleted from DNA
    
-   Copy Number Alteration: Sometimes a gene might get deleted, or copy-pasted into the genome multiple times. 
    
Once the differences are spotted, then the real fun begins! Bioinformaticians, biologists, computational biologists, clinicians and physicians are all interested in this data as these differences in DNA can often be the reason for a disease, can simply be a risk marker, or other times don't do anything at all.


## You mentioned data portals - where do data portals come into this?

The area of genomics has become so advanced that it's actually outpacing the very scientists who invented it! Since the cost of sequencing a genome has fallen so low, more and more projects are looking to sequencing, and there are even commercial companies that offer personal sequencing and clinical recommendations. Thats a lot of data! Where does all this data live, how do we protect it, and how do people look for it and access it? 

That’s where data portals come in! Data portals help build and manage the infrastructure to:

-   Store genomic data files    
-   Let people search through it (without actually touching the data)   
-   Download data  
-   Work with genomic data in cloud computing environments
-   Interact with data analytics in real-time analysis
    
We make it easy for scientists to find exactly what they need, and take care of hosting the data and its required infrastructure, saving them time and money so that they can focus on figuring out what the data is _actually_ telling us.

On top of that, data portals ingest and display some of the mutation data so that researchers can look for very specific cases. Its alot easier to browse these mutations on a webpage than downloading files and blindly searching through them one-by-one hoping to find the mutation of interest.


## Ok, so now you get what genomics is all about, but what’s so cool genomics?

There’s a couple really ideas that excite me about genomics. First, personalized medicine is a *really* hot topic. Research has shown that many different things affect how we get sick or react to treatments, from our actual DNA, to the environmental modifications on our DNA, or even the bacteria we have in our guts. Since each person has this unique biology, the goal of precision (or personalized) medicine enabled by genomics is to get to the point where doctors can offer treatment profiles based on your specific sgenetic profile. Cool, right?

Second, I just really like to know how biology works! Why do some people get cancer? What makes one person healthy and another person unhealthy? How does our environment affect our DNA and health? The big-data explosion of genomics helps us figure things like this out.

Lastly, we know that alot of this data is going to keep coming.  The first genome sequencing cost a billion dollars, but now a genome can be sequenced in a day for a cost of hundred of dollars (soon to be tens of dollars).  Genomic data is projected to exceed every other data source on this planet.  Who know's what cool stuff we will learn from this? I can't wait to find out!